x-pack/osquerybeat/internal/osqdcli/client.go

```diff
@@ -153,24 +179,47 @@ func (c *Client) close() {
 }
 
 // Query executes a given query, resolves the types
-func (c *Client) Query(ctx context.Context, sql string) ([]map[string]interface{}, error) {
+//
+// In order to workaround the issue https://github.com/elastic/beats/issues/36622
+// each query creates it's own RPC connection to osqueryd, allowing it to set a custom timeout per query.
+// Current implementation of osqueryd RPC returns the error when the long running query times out, but this timeout is a transport timeout,
+// that doesn't cancel the query execution itself.
+// This also makes the client RPC unusable until the long running query finishes, returning errors for each subsequent query.
```

Comment:
Can I confirm that the unusability is just the client and the server side is happy to take new connections in this situation?